jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/ResourceHttpContentFactory.java

```diff
@@ -82,29 +72,7 @@ private HttpContent load(String pathInContext, Resource resource)
         if (resource.isDirectory())
             return new ResourceHttpContent(resource, _mimeTypes.getMimeByExtension(resource.toString()));
```

Comment:
`Resource.toString()` is not implemented in `Resource`.
I would not assume "oh, but this is a PathResource so toString() is implemented`.
Plus `toString()` may be changed to a different format and totally break this code.
Why can it not be `pathInContext` like the line below?

Reply:
Not sure, this is pre-existing logic which has been maintained here.

So I'm not sure why directories need to be handled differently here, there is `resource.getFileName()` which is better than the `toString()`, but I also think `pathInContext` is good enough.

Reply:
I have now removed this the `resource.isDirectory` block.